### ISSUE TITLE
Move all ENV settings to UI

### ISSUE THREAD

**Issue by smilerz** [TandoorRecipes/recipes#1246]
In order to make Django easier to use and support:
- Move all or most ENV settings to be configurable in the UI.
- Ability to restart Django from UI
- Alert user when restart is required to change setting
- Provide visual indication\warning when http headers are not minimum required and/or are in conflict with settings.

**Comment by vabene1111** [comment]
interesting idea, would you write the env file from the ui and restart the container or how would you implement this ?

I really like this but we have to think about it carefully, no big project i can think of allow this and there might be a reason for that 🤔

**Comment by smilerz** [comment]
It looks like dotenv can do this.
https://saurabh-kumar.com/python-dotenv/reference/dotenv/#set_key

**Comment by smilerz** [comment]
this project is dead - but code can probably be borrowed to add ability to restart
https://bitbucket.org/lvlup/django-restart/src/master/

**Comment by vabene1111** [comment]
ok that would be an option. The one thing i could think of is security as the reason why other (even very large projects like grafana,. prometheus, ..) use .env files is security. 

But i think what they always do is have a page/service that checks the config for issues. Maybe we could add that to the system page to have an overview of all configured and not configured variables together with links to the docs (which i really really need to add)

**Comment by vabene1111** [comment]
this way we would also not need a restart function as apart from changing env vars a restart should never really be needed ... but we could also distinguish between security relevant and irrelevant settings. The irrelevant ones are configurable via UI the other not (this is how phabricator for example does it)

**Comment by smilerz** [comment]
> this way we would also not need a restart function as apart from changing env vars a restart should never really be needed ... but we could also distinguish between security relevant and irrelevant settings. The irrelevant ones are configurable via UI the other not (this is how phabricator for example does it)

I think keep any security related settings separate makes sense - though outside password and SECRET_KEY; are there many to worry about?

**Comment by vabene1111** [comment]
the allauth related stuff maybe and default group, email but most is not critical

**Comment by smilerz** [comment]
> the allauth related stuff maybe and default group, email but most is not critical
maybe adding some kind of 'test' button to validate that the configs work - but make password write only (never display)

**Comment by vabene1111** [comment]
that would still allow an possible attacker to change auth related things without the instance admin knowing. I think we need to be very careful here, there are certainly settigs where editing them from the UI would be cool but those could then also be stored in a DB, editing the .env file on the server seems like potential for very severe security issues.

**Comment by smilerz** [comment]
> that would still allow an possible attacker to change auth related things without the instance admin knowing. I think we need to be very careful here, there are certainly settigs where editing them from the UI would be cool but those could then also be stored in a DB, editing the .env file on the server seems like potential for very severe security issues.

I agree that it should be considered with care and to carefully consider what types of compromises can lead to what type of exposures.  

RE: the DB - my assumption has been that most of those settings are in the ENV as they are needed at startup before a DB is available. And I don't know that changing how the settings are stored does much one way or another about how secure it is, they both open exposure to changing settings surreptitiously.  The question is - what does someone gain, that already has elevated access, to changing any of those settings?

**Comment by vabene1111** [comment]
The idea of db was that someone would not have access to os level things (like environment variables) but just to a db.

There are some settings that, with proper caching, could be moved to a db instead of env.

I think we should look around a little how other projects treat this issue and then decide what to do.

**Comment by smaniktahla** [comment]
To piggyback on this, there are other containerized services that do this (IMO) better. I love Tandoor, but this is one area that is difficult to set up and troubleshoot.  

Here's a screenshot of calibre-web, which of course doesn't use Django, but does incorporate the email settings into an admin section of the application via the Web UI:

![Image](https://github.com/user-attachments/assets/d19fb7ea-3bd7-483b-a03c-ea9df2684edb)

Keeping these settings in the environment variables may make it more secure from a possible attack--though I also think that any attacker going after email settings is likely to do worse things than hijack my containered recipe server--but it makes it much harder to set up and troubleshoot, because no two applications use the same terminology for some pretty standard attributes to be passed along.

So I think there's always a bit of a tradeoff between security and useability, and in this case, I'm going to pick useability. 

I have to send my users a link to their invitation manually at the moment because the email I've set up in the container environment doesn't seem to be working. It's using Gmail with an app password, something I've done repeatedly for other services, but is almost impossible to troubleshoot without disrupting the service for anyone / everyone once it's set up because I have to change a variable and then spin up the entire stack, then try and send out a test email manually, then if nothing happens go back and edit the variable again...

![Image](https://github.com/user-attachments/assets/5a78c7e6-5c20-4ef8-b6da-a206e5920e0b)
